src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -1702,7 +1702,7 @@ void VM_Version::get_processor_features() {
   }
 
   // Use population count instruction if available.
-  if (supports_popcnt()) {
+  if (supports_popcnt() || supports_avx512_vpopcntdq() || supports_avx512_bitalg()) {
```

Comment:
The UsePopCountInstruction is only for scalar popcont. Extending it to Vector popcount is causing lot of confustion. Let us keep it for scalar only as below:
if (supports_popcnt()) {
 ...
}

Reply:
UsePopcountInstruction implies generating direct popcount instruction for targets which support it. A bit count operation can also be implemented without popcount as depicted by this patch.

Reply:
I agree its adding some confusion since user may turn off UsePopcountInstruction and may want to generate optimized instructions for bit count over AVX2 targets using Vector API.

For SLP  turning off UsePopcountInstruction will prevent creating scalar IR which will also inhabit interacting of vector operation by auto vectorizer.